assets/javascripts/modules/autoComplete.js

```diff
@@ -277,14 +284,20 @@ var isMatchingSuggestion = function (value) {
   if (value) {
     $(suggestions).each(function (index, suggestion) {
       if (value.toLowerCase() === suggestion.title.toLowerCase()) {
-        updateInput(suggestion.title, suggestion.value);
-        matchFound = true;
-        return false;
+        if ($autoCompleteInputElem.data('suggestionValue').toString() === suggestion.value) {
+          matchFound = true;
+          return false;
```

Comment:
Maybe just preference - you could simplify it by having a default var of true `var returnValue = true` and then set that to `false` in the conditionals and then have a return `returnValue` at the end.